### DISCUSSION TITLE
"Pending Review" Status

### DISCUSSION THREAD

**Discussion by rulim34** [payloadcms/payload#612] [Feature Requests & Ideas]
I would like to know, if there was a way to create feature like wordpress' pending review.

So the idea is, some user only allowed to add entry to collection, but the cannot publish it, they can only "submit" it to the "pending" status. It will be reviewed by other user with higher role/capability, which then will determine whether to publish it or not.

If it is currently impossible, could this feature be added in the future?

**Comment by jmikrut** [comment]
This is certainly possible right now using access control and hooks. It could even be built as a plugin so that the logic could be easily abstracted and shared between projects.

Due to Payload's open-ended access control, this might be hard to build into Payload itself given Payload does not know anything about your users or what they can / can't do. That's a good thing all things considered though.

@jacobsfletch and I have built a similar "pending review" system into a few of our own projects though. We may be able to abstract it into a plugin or similar, or at the very least, we could share our code with you to look at.

**Comment by endlessuniverse** [comment]
@jmikrut Please share it 🙏
Does it work with updates too?
Thanks!

**Comment by endlessuniverse** [comment]
On second thought, drafts are enough for now. Is it possible to disable publishing both on create and update for non admin users?

**Comment by andrewjroberts** [comment]
```
export const PublishingFields: Field[] = [
  {
    name: "editors",
    type: "relationship",
    relationTo: "users",
    hasMany: true,
    access: {
      create: isEditor,
      update: isEditor,
    },
    admin: {
      position: "sidebar",
    },
  },
  {
    name: "approvals",
    type: "relationship",
    relationTo: "users",
    hasMany: true,
    access: {
      create: isReviewer,
      update: isReviewer,
    },
    admin: {
      position: "sidebar",
    },
  },
];

export const ScheduledContentPublishingFields: Field[] = [
  // publishDate format for scheduled publishes
  {
    name: "publishDate",
    type: "date",
    access: {
      read: () => true,
      create: isReviewer,
      update: isReviewer,
    },
    admin: {
      position: "sidebar",
      description: "Post will go live on this date",
    },
    defaultValue: () => new Date(),
  },
  // extending base fields
  ...PublishingFields,
];

```

hey @endlessuniverse , this is my solution for that. restrict a separate "approvals" field and then check if that field is present in order to set a publish date.  let me know thoughts / questions

**Reply by endlessuniverse** [reply]
Thank you!

**Comment by ssyberg** [comment]
@jmikrut 
> Due to Payload's open-ended access control, this might be hard to build into Payload itself given Payload does not know anything about your users or what they can / can't do. That's a good thing all things considered though.

Another bump here, a very simple solution would be to add a new pending/moderation `_state` to core (that could be optionally enabled in config) which would have some implications in the UX (e.g. instead of `Save Draft` and `Publish` you might have `Save Draft` and `Submit for Review` and once in `pending` that button becomes `Publish`). Payload doesn't have to be opinionated or aware of our users or roles, the rest I think can be handles in access control/hooks.

**Reply by ssyberg** [reply]
Actually now that I'm looking deeper at the docs, I do think even putting aside a `pending` state there is some missing access control in core. I would expect a `publish` access control to be exposed already...

**Reply by DanRibbens** [reply]
I've been thinking about how _status options should be extensible to support customized publishing workflows. I'm with you.

**Reply by ssyberg** [reply]
Nice and I see this is already on the roadmap! https://github.com/payloadcms/payload/discussions/1009
